app/services/stockit/item_detail_sync.rb

```diff
@@ -0,0 +1,41 @@
+require 'stockit/base'
+
+module Stockit
+  class ItemDetailSync
+
+    include Stockit::Base
+
+    attr_accessor :detail, :detail_type
+
+    def initialize(detail)
+      @detail = detail
+      @detail_type = detail.class.name.underscore
+    end
+
+    class << self
+      def create(detail)
+        new(detail).create if detail
+      end
+
+      def update(detail)
+        new(detail).update if detail
+      end
+    end
+
+    def create
+      url = url_for("/api/v1/#{detail_type.pluralize}")
+      post(url, detail_params)
+    end
+
+    def update
+      url = url_for("/api/v1/#{detail_type.pluralize}/update")
+      put(url, detail_params)
+    end
+
+    protected
+
+    def detail_params
+      { "#{detail_type}": detail.attributes.except("id", "stockit_id" "created_at", "updated_at", "updated_by_id") }
```

Comment:
a comma is missing between `stcokit_id` and `created_at`.

Reply:
Good catch 👍 Thanks @namrataukirde . :)

Reply:
Why does it not generate an error ?
I miss statically compiled languages 😅